### PR TITLE
[GPU] Support empty tensor

### PR DESCRIPTION
**Pull request by yeonbok**
### Details:
 - Fixes to support empty tensor in gpu plugin 
 - Fixes bug in is_shape_agnostic in kernel param 

### Tickets:
 - 98121

### CODE REVIEW

File: src/plugins/intel_gpu/tests/test_cases/empty_tensor_gpu_test.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+using namespace cldnn;
+using namespace ::tests;
+namespace {
+struct empty_tensor_test_params {
+    layout nonzero_input_layout;
+    layout concat_input_layout;
+    int64_t concat_axis;
+};
+
+//void test_empty_tensor_concat_two_inputs(layout nonzero_input_layout, layout concat_input_layout, int32_t concat_axis) {
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/tests/test_cases/empty_tensor_gpu_test.cpp:26]: Please remove unnecessary remaining code

Comment (yeonbok) [reply src/plugins/intel_gpu/tests/test_cases/empty_tensor_gpu_test.cpp:26]: Fixed

**Comment by andrew-k-park** [comment]
Could you check segfault issue from quantize.dynamic TC for ov_gpu_unit_tests (https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/ie/job/ie-tests-linux-ubuntu20-cldnn_unit/39967/console)?

File: src/plugins/intel_gpu/src/plugin/infer_request.cpp [hunk 2]
```diff
@@ -563,9 +576,9 @@ void InferRequest::wait() {
             {
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/plugin/infer_request.cpp:576]: Can we combine `outputMemory != nullptr` check with `if (!bptr->is<gpu::ClBlob>()) {`?

File: src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp [hunk 3]
```diff
@@ -314,6 +320,14 @@ struct typed_primitive_impl_ocl : public typed_primitive_impl<PType> {
             _kernel_data.kernels[i].code.kernelString.reset();
         }
     }
+
+    void set_skip_kernels() {
```
[the remark this answers is not in the document]
Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp:324]: I prefer update_skip_kernels_list. Or how about update_kernels_list_to_skip ?

File: src/plugins/intel_gpu/src/kernel_selector/kernel_selector.cpp [hunk 4]
```diff
@@ -86,6 +86,12 @@ KernelsData kernel_selector_base::GetNaiveBestKernel(const KernelList& all_impls
             if (kds.size() && kds[0].kernels.size()) {
                 kernelsData = kds;
                 kernelName = implementation->GetName();
+                if (!params.is_shape_agnostic) {
```
[the remark this answers is not in the document]
Comment (yeonbok) [reply src/plugins/intel_gpu/src/kernel_selector/kernel_selector.cpp:89]: Seems that not all kernels are calling that function. Maybe we can refactor all kernels to use it, but we can do it in a separate work.

File: src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (sshlyapn) [reply src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp:324]: `update_kernels_list_to_skip` sounds good for me!

[the remark this answers is not in the document]
Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/impls/ocl/primitive_base.hpp:324]: Applied,

File: src/plugins/intel_gpu/src/plugin/infer_request.cpp (hunk 2 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/src/plugin/infer_request.cpp:570]: Not sure this is what you meant, but I did some more refactoring. https://github.com/openvinotoolkit/openvino/pull/15631/commits/70499d41d5efb9723c59b4c47749db23d0416418

Comment (sshlyapn) [reply src/plugins/intel_gpu/src/plugin/infer_request.cpp:570]: My suggestion was to extend [this condition](https://github.com/openvinotoolkit/openvino/blob/70499d41d5efb9723c59b4c47749db23d0416418/src/plugins/intel_gpu/src/plugin/infer_request.cpp#L568) with additional `outputMemory != nullptr` and remove it from [line 573](https://github.com/openvinotoolkit/openvino/blob/70499d41d5efb9723c59b4c47749db23d0416418/src/plugins/intel_gpu/src/plugin/infer_request.cpp#L573) and [line 575](https://github.com/openvinotoolkit/openvino/blob/70499d41d5efb9723c59b4c47749db23d0416418/src/plugins/intel_gpu/src/plugin/infer_request.cpp#L575)

Comment (yeonbok) [reply src/plugins/intel_gpu/src/plugin/infer_request.cpp:570]: Ah, I see. I updated : )

**Review by andrew-k-park — APPROVED** [review]

**Pull request merged**